pkg/apis/cluster/v1alpha1/clustercurator_types.go

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2020 Red Hat, Inc.
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClusterCurator is the Custom Resource object which holds the desired state and current status
+// of an cluster. ClusterCurator is a namespace scoped resource
+type ClusterCurator struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// spec holds desired configuration to create the job
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ClusterCuratorSpec `json:"spec"`
+
+	// status holds the information about the state of a cluster curator resource, managed cluster heealth and cluster health.  It is consistent with status information across
+	// the Kubernetes ecosystem.
+	// +optional
+	Status ClusterCuratorStatus `json:"status"`
+}
+
+// ClusterCuratorSpec defines the desired configuration to ceate a job
+type ClusterCuratorSpec struct {
+
+	// Desired action: create/import/destroy/detach/noop
+	// action holds the information about create/import/destroy/detach/noop one of the action job should perform
+	// +required
+	Action Action `json:"action,omitempty"`
+
+	// PreJobs holds the job specifications
+	// +optional
+	PreJobs []JobSpec `json:"prejobs"`
+
+	// PostJobs holds the job specifications
+	// +optional
+	PostJobs []JobSpec `json:"postjobs"`
+
+	// Jobs holds the job specifications
+	// +optional
+	Jobs []JobConfig `json:"jobs"`
```

Comment:
Should be JobSpec.... I guess in the yaml the Job was missing the RunOnAction

Reply:
it is missing type and RunOnAction. so I removed those. But I guess we can use same `JobSpec` here.

Reply:
yes JobConfig and JobSpec are the same and so we can drop JobSpec